### PR TITLE
New autoassign parameter for agent category belonging

### PR DESCRIPTION
**Pull request by xaviqv**
This update adds a new parameter "autoassign" that it's stored in the agent_category pivot table and it's meant to solve our particular scenario, where I have at the moment two categories:

1. The first one, is a category where I have one agent that has to check all new tickets:
- If it's not for her, or when It's needed, she has to be able to assign that ticket to other agents in the category (me or my boss)

2. The second one is the IT main category:
- We want to receive new tickets notification in one of our department shared mail accounts, so everyone can be noticed about new tickets
- Later, we want to distribute these tickets manually to whom is the best agent on each case, at least for the first support step

I've improved eloquent queries a bit. If it's worth to publish this then I'll create new translation fields and apply on the modified blade template

Also I've been thinking that a related configuration parameter would be nice to enable or not this behaviour.

Comments and suggestions are all welcome!

### CODE REVIEW

File: src/Views/admin/agent/index.blade.php [hunk 1]
```diff
@@ -83,7 +129,7 @@
                 @endforeach
                 </tbody>
             </table>
-
         @endif
     </div>
+	<script src="{!! asset('vendor/ticketit/js/agent_index.js') !!}"></script>
```
Comment (thekordy) [inline src/Views/admin/agent/index.blade.php]: Use instead:
```blade
@section('footer')
    <script type="text/javascript">
        $(function(){
            $('.jquery_agent_cat').click(function(){
                var checked=$(this).hasClass('checked');
                if (checked){
                    $(this).removeClass('checked');
                    $('#'+$(this).attr('id')+"_auto").prop('checked',false).prop('disabled','disabled');
                } else {
                    $(this).addClass('checked');
                    $('#'+$(this).attr('id')+"_auto").prop('checked','checked').prop('disabled',false);
                }
            });
        });
    </script>
@endsection
```

File: src/Views/admin/agent/index.blade.php [hunk 2]
```diff
@@ -48,24 +48,70 @@
...
+										   {!! (count($agent->categories->whereIn('id',$agent_cat->id)) == 0) ? "disabled=\"disabled\"" : ""  !!}
+										   
+										   ></td>
+											</tr>
+										@endforeach
+										</tbody>
+									</table>
+								  </div>
+								  <div class="modal-footer">
+									<button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>
```
Comment (thekordy) [inline src/Views/admin/agent/index.blade.php:94]: Can't submit the form, please check it.

File: src/Views/admin/agent/index.blade.php [hunk 3]
```diff
@@ -48,24 +48,70 @@
...
                                                         ],
                                             ]) !!}
-                            @foreach(\Kordy\Ticketit\Models\Category::all() as $agent_cat)
-                                <input name="agent_cats[]"
-                                       type="checkbox"
-                                       value="{{ $agent_cat->id }}"
-                                       {!! ($agent_cat->agents()->where("id", $agent->id)->count() > 0) ? "checked" : ""  !!}
-                                       > {{ $agent_cat->name }}
+									<table class="table table-hover table-striped">
+										<thead><th>Category</th>
```
Comment (thekordy) [inline src/Views/admin/agent/index.blade.php]: No plain words please,  use the translation

File: src/Views/admin/agent/index.blade.php [hunk 4]
```diff
@@ -48,24 +48,70 @@
...
+							
+							<div class="modal fade" id="CategoriesPopupAgent{{ $agent->id }}" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+							  <div class="modal-dialog" role="document">
+								<div class="modal-content">
+								  <div class="modal-header">
+									<button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+									<h4 class="modal-title" id="myModalLabel">Agent categories</h4>
+								  </div>
+								  <div class="modal-body">
+								  <h3 class="pull-left">Agent: {{ $agent->name }}</h3>
```
Comment (thekordy) [inline src/Views/admin/agent/index.blade.php]: No plain words please, use the translation

File: src/Views/admin/agent/index.blade.php [hunk 5]
```diff
@@ -48,24 +48,70 @@
...
+							<button type="button" class="btn btn-info btn-sm" data-toggle="modal" data-target="#CategoriesPopupAgent{{ $agent->id }}">
+							  Edit
+							</button>
+							
+							<div class="modal fade" id="CategoriesPopupAgent{{ $agent->id }}" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+							  <div class="modal-dialog" role="document">
+								<div class="modal-content">
+								  <div class="modal-header">
+									<button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+									<h4 class="modal-title" id="myModalLabel">Agent categories</h4>
```
Comment (thekordy) [inline src/Views/admin/agent/index.blade.php]: No plain words please, use the translation

File: src/Views/admin/agent/index.blade.php [hunk 6]
```diff
@@ -48,24 +48,70 @@
...
                                             ]) !!}
-                            @foreach(\Kordy\Ticketit\Models\Category::all() as $agent_cat)
-                                <input name="agent_cats[]"
-                                       type="checkbox"
-                                       value="{{ $agent_cat->id }}"
-                                       {!! ($agent_cat->agents()->where("id", $agent->id)->count() > 0) ? "checked" : ""  !!}
-                                       > {{ $agent_cat->name }}
+									<table class="table table-hover table-striped">
+										<thead><th>Category</th>
+										<th>Active</th>
```
Comment (thekordy) [inline src/Views/admin/agent/index.blade.php]: No plain words please, use the translation

File: src/Views/admin/agent/index.blade.php [hunk 7]
```diff
@@ -48,24 +48,70 @@
...
+										<th>Autoassign enabled</th></thead>
```
Comment (thekordy) [inline src/Views/admin/agent/index.blade.php]: No plain words please, use the translation

**Review by thekordy — CHANGES_REQUESTED** [review]

**Comment by xaviqv** [comment]
I've done all the changes. The form couldn't be submitted without publishing the .js file. Sorry for not specifying that.

I knew that all translation fields were pending and I'm sorry If I didn't explain well the status of this pull request as it gave you extra work. Next time I hope to be more clear about pending changes before PR approval.

Thanks for you feedback!

**Comment by xaviqv** [comment]
@thekordy Is there any requested change pending in this PR or anything else that has to be edited? Have you checked it out after my updates? Thank you!

File: src/Views/admin/agent/index.blade.php (hunk 2 above)
Comment (xaviqv) [reply src/Views/admin/agent/index.blade.php:94]: Corrected on 28 Mar.

**Review by xaviqv — COMMENTED** [review]
I think there are no pending modifications after my updates

**Comment by thekordy** [comment]
not ready for production, sorry.

**Pull request closed without merging**